node/src/components/chain_synchronizer/operations.rs

```diff
@@ -1005,33 +1006,55 @@ async fn fetch_forward(
 async fn fetch_to_genesis(trusted_block: &Block, ctx: &ChainSyncContext<'_>) -> Result<(), Error> {
     let _metric = ScopeTimer::new(&ctx.metrics.chain_sync_fetch_to_genesis_duration_seconds);
 
-    let locally_available_block_range = ctx
-        .effect_builder
-        .get_available_block_range_from_storage()
-        .await;
+    info!(
+        trusted_block_height = %trusted_block.height(),
+        locally_available_block_range_on_start = %ctx.locally_available_block_range_on_start,
+        "starting fetch to genesis",
+    );
 
     let mut walkback_block = trusted_block.clone();
     loop {
         // The available range from storage indicates a range of blocks for which we already have
         // all the corresponding deploys and global state stored locally.  Skip fetching for such a
         // range.
-        if locally_available_block_range.contains(walkback_block.height()) {
+        if ctx
+            .locally_available_block_range_on_start
+            .contains(walkback_block.height())
+        {
             let maybe_lowest_available_block = ctx
                 .effect_builder
-                .get_lowest_available_block_from_storage()
+                .get_block_at_height_with_metadata_from_storage(
```

Comment:
I see there is no `get_block_at_height_from_storage` storage request, but we could implement it to avoid unnecessary fetching of metadata, such as this particular case. As far as I can tell, this wouldn't be a breaking change.

Reply:
Yes, that's true.  I'll ticket it for a separate PR.